python/cudf/dataframe/series.py

```diff
@@ -308,7 +310,7 @@ def take(self, indices, ignore_index=False):
         if self.dtype == np.dtype("object"):
             return self[indices]
 
-        data = cudautils.gather(data=self.data.to_gpu_array(), index=indices)
+        col = cpp_copying.apply_gather_array(self.data.to_gpu_array(), indices)
```

Comment:
Gather the column instead of the array, then don't need to gather the mask separately.